main.go

```diff
@@ -484,6 +498,11 @@ func (s *SuperAgent) End(callback ...func(response Response, body string, errs [
 	}
 	req.URL.RawQuery = q.Encode()
 
+	// Add basic auth
+	if s.BasicAuth != nil {
```

Comment:
@parnurzeal I made this a pointer so I could check if it is nil here. I could change it to an empty struct though if you like